libcxx/include/__expected/expected.h

```diff
@@ -955,8 +956,17 @@ class expected {
     _LIBCPP_NO_UNIQUE_ADDRESS _ErrorType __unex_;
   };
 
+  _LIBCPP_HIDE_FROM_ABI static constexpr auto __calculate_padding() {
+    struct __calc_expected {
+      _LIBCPP_NO_UNIQUE_ADDRESS __union_t<_Tp, _Err> __union_;
+      bool __has_val_;
+    };
+    return sizeof(__calc_expected) - __libcpp_datasizeof<__calc_expected>::value;
```

Comment:
Why can't we just use `sizeof(expected) - __libcpp_datasizeof<expected>::value`?

Reply:
At this point the full definition of `expected` is not available yet, so I'm getting errors like:

```
__expected/expected.h:960:12: error: invalid application of 'sizeof' to an incomplete type 'expected<TracedBase<false, false>,
 TracedBase<true>>'                                                                                                                                                                             
# |   960 |     return sizeof(expected) - __libcpp_datasizeof<expected>::value;                                                                                                                 
# |       |            ^~~~~~~~~~~~~~~~                                           
```

Reply:
It seems we can determine the size of the padding via only `sizeof(__union_t<_Tp, _Err>)` and `__libcpp_datasizeof<__union_t<_Tp, _Err>>::value` without inventing a new type.